guide/results-viewer-react/src/components/YamlUrls/index.tsx

```diff
@@ -308,3 +271,126 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
 }
 
 export default Urls;
+
+interface RequestDetailsTabsProps extends HeadersViewProps, ResponseViewProps {}
+
+function RequestDetailsTabs ({ id, headersList, removeHeader, changeHeader, addHeader, responseCode, response }: RequestDetailsTabsProps): JSX.Element {
+  type tabType = "Headers" | "Response"
+  const [activeTab, setActiveTab] = useState<tabType>("Headers");
+  const tabs: tabType[] = ["Headers", "Response"];
+
+  return (
+    <div>
+      <div role="tablist" className="tab-list">
+        {tabs.map((tab) => (
+          <button
+            key={tab}
+            role="tab"
+            aria-selected={activeTab === tab}
+            aria-controls={`tabpanel-${tab}`}
+            id={`tab-${tab}`}
+            onClick={() => setActiveTab(tab)}
+            className={`tab ${activeTab === tab ? "active" : ""}`}
+            disabled={activeTab === tab}
+            style={{width: `${100 / tabs.length}%`}}
+          >
+            {tab}
+          </button>
+        ))}
+      </div>
+      <div role="tabpanel" id={`tabpanel-${activeTab}`} aria-labelledby={`tab-${activeTab}`}>
+        {activeTab === "Headers" && <HeadersView id={id} headersList={headersList} removeHeader={removeHeader} changeHeader={changeHeader} addHeader={addHeader} />}
+        {activeTab === "Response" && <ResponseView responseCode={responseCode} response={response} />}
+      </div>
+    </div>
+  );
+}
+
+interface HeadersViewProps {
+  id: string;
+  headersList: PewPewHeader[];
+  removeHeader: (headerId: string) => void;
+  changeHeader: (headerIndex: number, type: PewPewHeaderStringType, value: string) => void;
+  addHeader: (headerType?: HeaderType) => void;
+}
+
+function HeadersView ({ id, headersList, removeHeader, changeHeader, addHeader }: HeadersViewProps): JSX.Element {
```

Comment:
When creating sub elements, it's a bit cleaner to put them in their own file and create a separate storybook for them.